webapp/src/dogma/features/token/NewToken.tsx

```diff
@@ -0,0 +1,141 @@
+import {
+  Button,
+  Checkbox,
+  Flex,
+  FormControl,
+  FormErrorMessage,
+  FormLabel,
+  Input,
+  Popover,
+  PopoverArrow,
+  PopoverBody,
+  PopoverCloseButton,
+  PopoverContent,
+  PopoverFooter,
+  PopoverHeader,
+  PopoverTrigger,
+  Spacer,
+  useDisclosure,
+} from '@chakra-ui/react';
+import { SerializedError } from '@reduxjs/toolkit';
+import { FetchBaseQueryError } from '@reduxjs/toolkit/dist/query';
+import { DisplaySecretModal } from 'dogma/features/token/DisplaySecretModal';
+import { useAddNewTokenMutation } from 'dogma/features/api/apiSlice';
+import { createMessage } from 'dogma/features/message/messageSlice';
+import ErrorHandler from 'dogma/features/services/ErrorHandler';
+import { useAppDispatch, useAppSelector } from 'dogma/store';
+import { useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { IoMdArrowDropdown } from 'react-icons/io';
+
+const APP_ID_PATTERN = /^[0-9A-Za-z](?:[-+_0-9A-Za-z\.]*[0-9A-Za-z])?$/;
+
+type FormData = {
+  appId: string;
+  isAdmin: boolean;
+};
+
+export const NewToken = () => {
+  const {
+    isOpen: isNewTokenFormOpen,
+    onToggle: onNewTokenFormToggle,
+    onClose: onNewTokenFormClose,
+  } = useDisclosure();
+  const {
+    isOpen: isSecretModalOpen,
+    onToggle: onSecretModalToggle,
+    onClose: onSecretModalClose,
+  } = useDisclosure();
+  const {
+    register,
+    handleSubmit,
+    reset,
+    formState: { errors },
+  } = useForm<FormData>();
+  const [addNewToken, { isLoading }] = useAddNewTokenMutation();
+  const [tokenDetail, setTokenDetail] = useState(null);
+  const dispatch = useAppDispatch();
+  const { user } = useAppSelector((state) => state.auth);
+  const onSubmit = async (formData: FormData) => {
+    const data = `appId=${formData.appId}&isAdmin=${formData.isAdmin}`;
```

Comment:
Should we fall back to `false` if `isAdmin` is not set?
Otherwise, `undefined` is submitted. 
```suggestion
    const data = `appId=${formData.appId}&isAdmin=${formData.isAdmin || false}`;
```
<img width="462" alt="image" src="https://user-images.githubusercontent.com/1866157/217984869-8184b436-ced2-48c3-a6ba-9df0d6759d4b.png">

Reply:
<img width="1166" alt="Screen Shot 2023-02-10 at 10 41 28" src="https://user-images.githubusercontent.com/5079588/217995053-c9dc63fc-4124-438e-bd0c-5ba27ded2429.png">